lib/rbs/json_schema/generator.rb

```diff
@@ -0,0 +1,453 @@
+      def generate
+        # Validate options received from CLI
+        validate_options()
+
+        # Read filename & it's contents
+        file_path do |uri|
+          stdout.puts "\n======= Generating RBS for schema: #{uri.path} =======\n\n"
+
+          @decls << RBS::AST::Declarations::Module.new(
+            name: generate_type_name_for_uri(uri, module_name: true),
+            type_params: [],
+            members: [],
+            self_types: [],
+            annotations: [],
+            location: nil,
+            comment: nil
+          )
+          generate_rbs(uri, read_from_uri(uri))
+
+          # Write output to either a file or STDOUT
+          write_output()
+          # Reinitialize declarations array to default value, since the declarations have successfully been written to an IO
+          @decls = []
+
+          stdout.puts "\n\n======= Completed generating RBS for schema: #{uri.path} =======\n"
+        end
+      end
+
+      # IMPORTANT: Function used to generate AST alias declarations from a URI & a schema document
+      def generate_rbs(uri, document)
+        # If schema is already generated for a URI, do not re-generate declarations/types
+        if fragment = uri.fragment
+          return if fragment.empty? # If fragment is empty, implies top level schema which is always generated since it is the starting point of the algorithm
+
+          if !generated_schemas.dig(uri.path, fragment).nil? # Check if types have been generated for a particular path & fragment
+            return
+          end
+        else
+          if !generated_schemas.dig(uri.path).empty? # Check if types have been generated for a particular path
+            return
+          end
+        end
+
+        unless document.is_a?(Hash)
+          raise ValidationError.new(message: "Invalid JSON Schema: #{document}")
+        end
+
+        if fragment = uri.fragment
+          generated_schemas[uri.path][fragment.empty? ? "#" : fragment] = true
+        else
+          generated_schemas[uri.path] = {}
+        end
+        # Parse & generate declarations from remaining schema content
+        decl = Alias.new(
+          name: generate_type_name_for_uri(uri), # Normal type name with no prefix
+          type: translate_type(uri, document), # Obtain type of alias by parsing the schema document
+          annotations: [],
+          location: nil,
+          comment: nil
+        )
+        # Append the declaration if & only if the declaration has a valid RBS::Type assigned
+        @decls.first.members << decl if !decl.type.nil?
+      end
+
+      # Parse JSON schema & return the `RBS::Types` to be assigned
+      def translate_type(uri, schema)
+        # Default initial value
+        type = nil
+
+        # Check for various schema attributes presence
+        case
+        # SCHEMA_DEFINITIONS[3] => enum => A union of literals
+        when schema.key?(SCHEMA_DEFINITIONS[3])
```

Comment:
The implementation for `enum` type would be something like the following, without `SCHEMA_DEFINITIONS` constant and `fetch_type` method:

```rb
        when values = schema[:enum]
          unless values.is_a?(Array)
            raise ValidationError.new(......)
          end

          types = values.map do |value|
            case value
            when String, Integer
              Types::Literal.new(literal: value, location: nil)
            when nil
              Types::Bases::Nil.new(location: nil)
            else
              raise ......
            end
          end

          type = Types::Union.new(types: types, location: nil)
```

I believe this is short and understandable enough.